autoprotocol/protocol.py

```diff
@@ -5080,51 +5081,62 @@ def provision(self, resource_id, dests, volumes):
         dests = WellGroup(dests)
         if not isinstance(resource_id, str):
             raise TypeError("Resource ID must be a string.")
-        if not isinstance(volumes, list):
-            volumes = [Unit(volumes)] * len(dests)
+        if not isinstance(amounts, list):
+            amounts = [Unit(amounts)] * len(dests)
         else:
-            if len(volumes) != len(dests):
+            if len(amounts) != len(dests):
                 raise RuntimeError(
                     "To provision a resource into multiple "
-                    "destinations with multiple volumes, the  "
-                    "list of volumes must correspond with the "
+                    "destinations with multiple volumes/masses, the  "
+                    "list of volumes/masses must correspond with the "
                     "destinations in length and in order."
                 )
-            volumes = [Unit(v) for v in volumes]
-        for v in volumes:
-            if not isinstance(v, (str, Unit)):
-                raise TypeError("Volume must be a string or Unit.")
+            amounts = [Unit(v) for v in amounts]
+        unique_measure_modes = set()
+        for amount in amounts:
+            if not isinstance(amount, (str, Unit)):
+                raise TypeError("Volume/Mass must be a string or Unit.")
+            if amount.unit in VOLUMETRIC_UNITS:
+                unique_measure_modes.add("volume")
+            elif amount.unit in MASS_UNITS:
+                unique_measure_modes.add("mass")
```

Comment:
hmm we should probably explicitly fail here instead of silently fail? (i.e. else case)

Reply:
Actually below code will catch(I know message is not capturing that).
```
        if len(unique_measure_modes) != 1:
            raise ValueError("Received amounts with more than one measurement mode")
```

Anyhow I added else block.